### PR TITLE
Add BASH auto completion

### PR DESCRIPTION
**Pull request by sunwire**

### CODE REVIEW

File: mt-st [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+_mt () {
+    local cur prev words cword
+    _init_completion || return
+
+    #possible commands
+    commands="weof wset eof fsf fsfm bsf bsfm fsr bsr fss bss rewind offline rewoffl eject retension eod seod seek tell status erase setblk lock unlock load compression setdensity drvbuffer stwrthreshold stoptions stsetoptions stclearoptions defblksize defdensity defdrvbuffer defcompression stsetcln sttimeout stlongtimeout densities setpartition mkpartition partseek asf stshowopt"
+
+    if [ -x "$(command -v udevadm)" ]; then
+        #list tape devices
+        for d in `udevadm trigger --verbose --dry-run --type=devices --subsystem-match=scsi_tape`; do
```
Comment (iustin) [inline mt-st]: This will re-run udevadm at each tab, not only when prev is `-f`. How about moving this into the `-f` block blow?

File: mt-st [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#
+
+_mt () {
+    local cur prev words cword
+    _init_completion || return
+
+    #possible commands
+    commands="weof wset eof fsf fsfm bsf bsfm fsr bsr fss bss rewind offline rewoffl eject retension eod seod seek tell status erase setblk lock unlock load compression setdensity drvbuffer stwrthreshold stoptions stsetoptions stclearoptions defblksize defdensity defdrvbuffer defcompression stsetcln sttimeout stlongtimeout densities setpartition mkpartition partseek asf stshowopt"
+
+    if [ -x "$(command -v udevadm)" ]; then
```
Comment (iustin) [inline mt-st]: Does this mean that only when udevadm is installed, we can complete tape devices? Or do I misunderstand here?

Comment (sunwire) [reply mt-st]: Yes. udevadm is run only when is installed,

File: mt-st (hunk 1 above)
Comment (sunwire) [reply mt-st]: You are right. It will be a better place for this code.

**Comment by sunwire** [comment]
Sorry for the little mess. I pasted it wrong.
I added _filedir when udevadm is unavailable
If you want I can cancel this PR and make a new one or delete last 2 commits (rebase)

**Pull request closed without merging**